match: inline in expression / compiler error: use of undeclared identifier

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
OS: macos, macOS, 11.4, 20F71
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i7-6567U CPU @ 3.30GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.9)

getwd: /Users/ah/SVN-Checkouts/TST/vlang-test
vmodules: /Users/ah/.vmodules
vroot: /usr/local/Cellar/vlang/0.2.2/libexec
vexe: /usr/local/Cellar/vlang/0.2.2/libexec/v
vexe mtime: 2021-06-06 21:30:18
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 f4486d7.114c382

Git version: git version 2.31.1
Git vroot status: weekly.2021.22-52-g114c382c
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02 

**What did you do?**

Used a `match` as inline function to check values based on sum types.
The problem **only exists** if there is **more than one** `match {..} || match {...}` in the expression

**What did you expect to see?**

no failed compilation
 
**What did you see instead?**

```
v -cg run types-match.v
/tmp/v/types-match.15099181759772172290.tmp.c:10116:26: error: use of undeclared identifier '_t734'; did you mean '_t733'?
        }       bool res = !(_t733 || _t734);
                                      ^~~~~
                                      _t733
/tmp/v/types-match.15099181759772172290.tmp.c:10100:7: note: '_t733' declared here
        bool _t733;
             ^
1 error generated.
```

starting at 10099:
```c
bool main__isok(main__SumType a, main__SumType b) {
	bool _t733; /* ERROR LINE 10100 */
	if (a._typ == 7 /* int */) {
		_t733 = ((b._typ == 7 /* int */) ? ((*a._int) == (*b._int)) : (false));
		;
	}
	else if (a._typ == 84 /* main.Empty */) {
	bool _t734;
	if (a._typ == 7 /* int */) {
		_t734 = ((b._typ == 7 /* int */) ? ((*a._int) + 10 == (*b._int) - 10) : (false));
		;
	}
	else if (a._typ == 84 /* main.Empty */) {
		_t734 = false;
		;
	}		_t733 = false;
		;
	}	bool res = !(_t733 || _t734);  /* ERROR LINE 10116 */
	return res;
}
```

`source.v`
```v
struct Empty {}

type SumType = Empty | int

fn isok(a SumType, b SumType) bool {
    return !(match a {
        int {
            match b {
                int { a == b }
                Empty { false }
            }
        }
        Empty {
            false
        }
    } || match a {
        int {
            match b {
                int { a + 10 == b - 10 }
                Empty { false }
            }
        }
        Empty {
            false
        }
    })
}

a := SumType(1)
b := SumType(Empty{})

res := isok(a, b)

dump(res)
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4127,7 +4127,6 @@ fn (mut g Gen) match_expr_sumtype(node ast.MatchExpr, is_expr bool, cond_var str
 					if is_expr && tmp_var.len == 0 {
 						g.write(' : ')
 					} else {
-						g.writeln('')
 						g.write_v_source_line_info(branch.pos)
 						g.write('else ')
 					}
@@ -4168,7 +4167,8 @@ fn (mut g Gen) match_expr_sumtype(node ast.MatchExpr, is_expr bool, cond_var str
 			g.stmts_with_tmp_var(branch.stmts, tmp_var)
 			g.expected_cast_type = 0
 			if g.inside_ternary == 0 {
-				g.write('}')
+				g.writeln('}')
+				g.stmt_path_pos << g.out.len
 			}
 			sumtype_index++
 			if branch.exprs.len == 0 || sumtype_index == branch.exprs.len {

--- a/vlib/v/tests/match_with_multi_sumtype_exprs_test.v
+++ b/vlib/v/tests/match_with_multi_sumtype_exprs_test.v
@@ -1,0 +1,37 @@
+struct Empty {}
+
+type SumType = Empty | int
+
+fn isok(a SumType, b SumType) bool {
+	return !(match a {
+		int {
+			match b {
+				int { a == b }
+				Empty { false }
+			}
+		}
+		Empty {
+			false
+		}
+	} || match a {
+		int {
+			match b {
+				int { a + 10 == b - 10 }
+				Empty { false }
+			}
+		}
+		Empty {
+			false
+		}
+	})
+}
+
+fn test_match_with_multi_sumtype_exprs() {
+	a := SumType(1)
+	b := SumType(Empty{})
+
+	res := isok(a, b)
+
+	dump(res)
+	assert res
+}

